### PR TITLE
Add k8s example to auth docs

### PR DESCRIPTION
**Pull request by flaviuvadan**
**Pull Request Checklist**
- [ ] Fixes #
- [ ] Tests added
- [x] Documentation/examples added
- [ ] [Good commit messages](https://cbea.ms/git-commit/) and/or PR title

Vikram pointed out in the [CNCF #hera channel](https://cloud-native.slack.com/archives/C03NRMD9KPY/p1695030320402139) that we no longer have a K8s auth example. This PR adds back the example to the docs

### CODE REVIEW

File: docs/walk-through/authentication.md [hunk 1]
```diff
@@ -100,6 +101,55 @@ class MyTokenGenerator(TokenGenerator):
...
+
+##### Generating a K8s token via `TokenGenerator`
+
+```python
+import base64
+import errno
+import os
+from typing import Optional
+
+from kubernetes import client, config
```
Comment (elliotgunton) [inline docs/walk-through/authentication.md:124]: This would require users to pip install kubernetes right? Could we make a note of that

**Review by elliotgunton — COMMENTED** [review]
LGTM but we should add a note about the new dependency that a user would need to install

File: docs/walk-through/authentication.md (hunk 1 above)
Comment (flaviuvadan) [reply docs/walk-through/authentication.md:124]: Sure thing!

**Review by elliotgunton — APPROVED** [review]
LGTM!

**Pull request merged**